### YamlMime:ManagedReference
items:
- uid: System.Xaml.XamlInternalException
  id: XamlInternalException
  children:
  - System.Xaml.XamlInternalException.#ctor
  - System.Xaml.XamlInternalException.#ctor(System.String)
  - System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: XamlInternalException
  nameWithType: XamlInternalException
  fullName: System.Xaml.XamlInternalException
  type: Class
  summary: "A exceção que é lançada para inconsistências internas que ocorrem durante a leitura de XAML e gravação de XAML."
  remarks: "Essa exceção geralmente é lançada quando algum aspecto do leitor XAML ou do sistema de gravador XAML foi estendido pelo seu código e inconsistências ou problemas de implementação ocorreram que foram passados para o código de sistema de serviços XAML do .NET Framework. Exceções que podem se originar de terceiros, como analisando o XAML e encontrar erros em XAML, não devem lançar XamlInternalException e, em vez disso, acionar outras exceções como <xref:System.Xaml.XamlParseException>ou <xref:System.Xaml.XamlObjectWriterException>.</xref:System.Xaml.XamlObjectWriterException> </xref:System.Xaml.XamlParseException>       Se você estiver escrevendo código que se estende serviços XAML do .NET Framework, em particular, se você estiver escrevendo código para um leitor XAML ou a implementação do gravador XAML, você deve manter essa distinção entre XamlInternalException e outras exceções relacionados a XAML. O código só deve lançar XamlInternalException em casos onde extensão adicional de sua classe viola alguma princípio ou implementação padrão de design de que define seu código pode ser chamado. Seu código deve lançar exceções outros XAML específicas para casos de exceção em que o leitor XAML ou a implementação do gravador XAML é usada sem a extensão e é usada para carregar o XAML de terceiros, o contexto do esquema XAML de terceiros e assim por diante."
  syntax:
    content: 'public class XamlInternalException : System.Xaml.XamlException'
  inheritance:
  - System.Exception
  - System.Xaml.XamlException
  implements: []
  inheritedMembers:
  - System.Xaml.XamlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Xaml.XamlException.LineNumber
  - System.Xaml.XamlException.LinePosition
  - System.Xaml.XamlException.Message
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor
  id: '#ctor'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Inicializa uma nova instância do <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> classe com uma mensagem fornecida pelo sistema que descreve o erro."
  syntax:
    content: public XamlInternalException ();
    parameters: []
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Inicializa uma nova instância do <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> classe com uma mensagem especificada que descreve o erro."
  syntax:
    content: public XamlInternalException (string message);
    parameters:
    - id: message
      type: System.String
      description: "A mensagem que descreve a exceção."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Inicializa uma nova instância do <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> classe com os dados serializados."
  syntax:
    content: protected XamlInternalException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "O objeto que contém os dados do objeto serializado."
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "As informações contextuais sobre a origem ou destino."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Xaml.XamlInternalException
  langs:
  - csharp
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
  type: Constructor
  assemblies:
  - System.Xaml
  namespace: System.Xaml
  summary: "Inicializa uma nova instância do <xref href=&quot;System.Xaml.XamlInternalException&quot;> </xref> classe com uma mensagem de erro especificada e uma referência à exceção interna que é a causa dessa exceção."
  syntax:
    content: public XamlInternalException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "A mensagem que descreve a exceção."
    - id: innerException
      type: System.Exception
      description: "A exceção que é a causa da exceção atual. Se o `innerException` parâmetro não é <xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;> </xref>, a exceção atual é gerada em um <xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;> </xref> bloco que manipula a exceção interna."
  overload: System.Xaml.XamlInternalException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.Xaml.XamlException
  isExternal: false
  name: System.Xaml.XamlException
- uid: System.Xaml.XamlInternalException.#ctor
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException()
  nameWithType: XamlInternalException.XamlInternalException()
  fullName: System.Xaml.XamlInternalException.XamlInternalException()
- uid: System.Xaml.XamlInternalException.#ctor(System.String)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String)
  nameWithType: XamlInternalException.XamlInternalException(String)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Xaml.XamlInternalException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(SerializationInfo,StreamingContext)
  nameWithType: XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Xaml.XamlInternalException.#ctor(System.String,System.Exception)
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException(String,Exception)
  nameWithType: XamlInternalException.XamlInternalException(String,Exception)
  fullName: System.Xaml.XamlInternalException.XamlInternalException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Xaml.XamlInternalException.#ctor*
  parent: System.Xaml.XamlInternalException
  isExternal: false
  name: XamlInternalException
  nameWithType: XamlInternalException.XamlInternalException
